### ISSUE TITLE
Malformed certificate file

### ISSUE THREAD

**Issue by anonimal** [monero-project/kovri#30]
Blocker. Bails when parsing j_at_torontocrypto.org.crt. I thought x509v3 extensions were to blame but parg_at_mail.i2p.crt works fine. Most likely yet another implementation issue.

**Comment by majestrate** [comment]
fails how?

**Comment by anonimal** [comment]
```
16:11:25/error - Malformed certificate file
16:11:25/error - Failed to load reseed certificates
16:11:25/error - Reseed failed
```

**Comment by anonimal** [comment]
```
X.509 Certificate Information:
	Version: 3
	Serial Number (hex): 269cc84d7e3421a72c60eaf75fdf203f
	Issuer: C=XX,O=I2P Anonymous Network,OU=I2P,L=XX,street=XX,CN=j@torontocrypto.org
	Validity:
		Not Before: Tue Sep 22 22:15:33 UTC 2015
		Not After: Mon Sep 22 22:15:33 UTC 2025
	Subject: C=XX,O=I2P Anonymous Network,OU=I2P,L=XX,street=XX,CN=j@torontocrypto.org
	Subject Public Key Algorithm: RSA
	Algorithm Security Level: High (4096 bits)
		Modulus (bits 4096):
			00:a6:d0:1f:ad:51:89:b0:1e:2d:17:a6:61:a8:7f:d3
			bd:70:3e:27:7d:f6:fa:4f:a1:5c:49:33:01:5c:ff:8d
			a4:6e:7a:52:14:75:22:5d:16:0f:58:c9:33:a8:f9:9d
			65:19:2b:f4:13:6a:b7:71:5a:d7:d2:22:31:06:c9:e3
			dc:8a:46:1f:7e:1c:f2:74:f5:bf:3c:af:66:7b:e7:0a
			3b:d5:be:1e:ab:83:a1:77:cf:f8:a5:27:e7:ef:33:c2
			c6:da:af:06:1e:fd:37:47:ba:3b:58:c2:ec:8d:35:5a
			45:da:41:f4:4d:d4:1a:48:a5:14:ff:f3:41:fb:9a:e5
			8e:1e:83:7f:00:56:c9:2f:62:8a:79:1c:7e:1b:58:6c
			1e:a9:f7:a4:17:2c:ab:94:15:f9:b0:10:76:9c:f1:61
			86:0a:1e:7f:c4:3c:56:24:f4:b3:37:a7:51:76:b9:63
			90:0c:dd:14:53:ea:c8:36:fa:70:24:84:7e:72:f9:3f
			6d:3a:4f:a4:c0:90:ad:82:17:76:7e:21:c6:00:97:ec
			a7:8a:62:fb:05:a5:38:9e:32:61:94:d2:89:85:19:63
			11:73:99:e6:db:74:44:a6:6d:4d:71:4e:ce:8f:c9:14
			21:6f:58:6c:23:33:47:82:93:ff:1e:a1:d4:1c:11:4b
			82:fd:1e:af:e8:7d:03:3e:bc:35:df:7f:7c:ef:73:0e
			cc:32:65:69:81:4f:e3:87:ff:48:5c:d7:3a:03:72:01
			1d:fa:49:6b:12:33:86:4e:d4:7c:e6:57:ea:4d:64:c9
			e5:e3:5c:0f:06:af:91:6f:03:d0:29:f5:e1:fb:29:51
			57:7b:26:98:ab:58:13:5a:b1:80:30:fb:e0:63:a7:4d
			00:7e:ce:15:7b:6a:65:41:ab:20:0a:b2:5a:76:da:c4
			4b:0e:d4:75:b2:45:88:ad:08:cf:75:2c:8e:60:15:44
			54:c1:03:f3:a3:67:b0:82:c8:6a:d2:e1:9a:3b:ed:89
			eb:57:cc:f5:c6:fc:68:a3:94:f1:86:9a:e0:dd:38:b1
			97:ee:f3:99:38:d3:62:ab:ed:b3:ff:0c:4e:c7:bf:d2
			b4:f9:4e:59:b5:b0:89:c6:b3:94:cf:96:20:6d:f0:ab
			87:f8:b0:16:d8:bc:54:ff:30:16:fd:bd:4b:57:7a:2a
			a8:fd:39:5a:e8:33:59:ec:f5:a3:70:ab:26:2c:31:b8
			ff:49:37:bc:04:14:67:73:1b:b0:96:47:fa:f4:55:f1
			c9:a1:ad:c9:f7:3d:af:2f:1e:9e:42:fc:41:8c:b3:6d
			50:d1:5f:0d:35:ca:88:30:4e:86:7c:31:cf:cf:8a:72
			3b
		Exponent (bits 24):
			01:00:01
	Extensions:
		Key Usage (critical):
			Digital signature.
			Certificate signing.
		Key Purpose (not critical):
			TLS WWW Client.
			TLS WWW Server.
		Basic Constraints (critical):
			Certificate Authority (CA): TRUE
		Subject Key Identifier (not critical):
			6a40746f726f6e746f63727970746f2e6f7267
		Authority Key Identifier (not critical):
			6a40746f726f6e746f63727970746f2e6f7267
	Signature Algorithm: RSA-SHA256
	Signature:
		91:a6:66:fd:d3:28:2c:2e:35:a7:e6:3d:40:af:bc:95
		7a:3f:07:41:a9:87:55:5f:6e:f4:a4:b2:36:22:33:da
		e6:c1:3e:74:38:82:ae:ba:c7:e6:d5:6c:a1:47:d8:d7
		10:32:d1:e7:25:89:cd:d2:0a:ed:83:75:0c:62:71:81
		48:b3:23:7d:d6:8c:99:5a:58:e7:56:e3:96:0d:14:28
		c9:e1:af:73:03:77:14:df:16:41:9f:d5:9e:85:ff:6f
		85:ed:c7:40:f7:1e:c4:23:2d:77:f3:93:40:33:49:7a
		7d:4c:ff:7f:a2:b3:11:e5:b9:16:8a:a8:54:85:7b:e5
		6f:b5:e1:a5:9f:a9:f7:ff:0d:d0:3e:db:8d:c6:89:ef
		84:67:ea:0d:ec:4b:ef:f0:28:9d:1c:ee:1e:51:df:e9
		91:34:f5:b5:49:e7:1b:7d:01:4d:fc:69:20:0d:0c:59
		b1:cb:e6:6d:1b:02:1b:4d:c8:45:07:98:a6:ab:76:33
		c2:80:e4:a5:ee:fc:7e:99:53:95:e5:0b:af:95:23:bb
		df:eb:25:96:3d:fd:ec:f0:24:41:0e:07:d8:b4:8e:ee
		44:59:1f:c0:92:ec:4c:16:42:38:5a:32:41:ef:85:08
		86:59:8f:89:60:e9:fe:3a:14:73:1e:91:b3:e8:bb:0f
		a0:d8:1f:b6:40:d7:56:f9:f6:6a:75:52:c0:5d:bb:8d
		db:13:2c:55:a9:19:9f:95:fd:88:17:da:9c:d9:8d:b7
		94:07:e1:3f:35:49:96:fd:64:82:52:0f:e2:f4:94:a8
		8c:c3:59:25:6d:a6:18:3e:9d:a6:8f:06:e8:e3:60:e1
		b1:ec:dc:25:7d:d3:29:78:4c:7b:7a:a4:61:9d:c8:13
		46:bd:2a:ae:13:9a:85:80:91:dc:9c:4f:5a:07:e7:5c
		26:c9:bc:16:73:78:b1:ee:b5:db:51:ef:21:e8:a8:01
		11:7a:89:57:ad:32:6d:7e:e2:69:ca:ea:f5:68:06:c4
		8a:f9:86:74:94:10:66:6d:31:c7:61:10:0e:28:0f:7d
		7d:87:ef:23:fd:48:d7:2a:05:b0:ba:e3:44:c6:94:a7
		01:16:f0:41:36:cf:27:f3:cb:0f:a9:90:56:91:ef:8c
		bd:92:f6:76:db:d6:bf:c6:99:17:73:0a:91:d5:c0:b1
		9c:cd:42:6f:c1:f8:c0:67:c6:c3:08:e3:52:18:7c:ee
		cf:0f:25:82:97:51:a5:d7:96:08:b7:d0:cd:7c:a8:9a
		38:74:53:c4:0d:a5:e1:c4:31:d4:21:9a:76:98:6b:c5
		89:f3:70:d3:9c:ee:e9:c1:aa:18:aa:37:83:43:dc:8c
Other Information:
	SHA1 fingerprint:
		5e2d511f4d554caa921c30d3a48c34b10b942654
	SHA256 fingerprint:
		7c33f036740eac66ea5c83f8a83c18243b9e7ea274a70e0e130c8e5e173ce487
	Public Key ID:
		a0c88a08c21346d7b1515d2849e3048045eec163
	Public key's random art:
		+--[ RSA 4096]----+
		| .+=o+==o o.     |
		|..+  .=o.o       |
		| o E ....        |
		|o.+.o. .         |
		|ooo..   S        |
		|=..              |
		|+                |
		|                 |
		|                 |
		+-----------------+

-----BEGIN CERTIFICATE-----
MIIF7TCCA9egAwIBAgIQJpzITX40IacsYOr3X98gPzALBgkqhkiG9w0BAQswczEL
MAkGA1UEBhMCWFgxHjAcBgNVBAoTFUkyUCBBbm9ueW1vdXMgTmV0d29yazEMMAoG
A1UECxMDSTJQMQswCQYDVQQHEwJYWDELMAkGA1UECRMCWFgxHDAaBgNVBAMME2pA
dG9yb250b2NyeXB0by5vcmcwHhcNMTUwOTIyMjIxNTMzWhcNMjUwOTIyMjIxNTMz
WjBzMQswCQYDVQQGEwJYWDEeMBwGA1UEChMVSTJQIEFub255bW91cyBOZXR3b3Jr
MQwwCgYDVQQLEwNJMlAxCzAJBgNVBAcTAlhYMQswCQYDVQQJEwJYWDEcMBoGA1UE
AwwTakB0b3JvbnRvY3J5cHRvLm9yZzCCAiIwDQYJKoZIhvcNAQEBBQADggIPADCC
AgoCggIBAKbQH61RibAeLRemYah/071wPid99vpPoVxJMwFc/42kbnpSFHUiXRYP
WMkzqPmdZRkr9BNqt3Fa19IiMQbJ49yKRh9+HPJ09b88r2Z75wo71b4eq4Ohd8/4
pSfn7zPCxtqvBh79N0e6O1jC7I01WkXaQfRN1BpIpRT/80H7muWOHoN/AFbJL2KK
eRx+G1hsHqn3pBcsq5QV+bAQdpzxYYYKHn/EPFYk9LM3p1F2uWOQDN0UU+rINvpw
JIR+cvk/bTpPpMCQrYIXdn4hxgCX7KeKYvsFpTieMmGU0omFGWMRc5nm23REpm1N
cU7Oj8kUIW9YbCMzR4KT/x6h1BwRS4L9Hq/ofQM+vDXff3zvcw7MMmVpgU/jh/9I
XNc6A3IBHfpJaxIzhk7UfOZX6k1kyeXjXA8Gr5FvA9Ap9eH7KVFXeyaYq1gTWrGA
MPvgY6dNAH7OFXtqZUGrIAqyWnbaxEsO1HWyRYitCM91LI5gFURUwQPzo2ewgshq
0uGaO+2J61fM9cb8aKOU8Yaa4N04sZfu85k402Kr7bP/DE7Hv9K0+U5ZtbCJxrOU
z5YgbfCrh/iwFti8VP8wFv29S1d6Kqj9OVroM1ns9aNwqyYsMbj/STe8BBRncxuw
lkf69FXxyaGtyfc9ry8enkL8QYyzbVDRXw01yogwToZ8Mc/PinI7AgMBAAGjgYAw
fjAOBgNVHQ8BAf8EBAMCAIQwHQYDVR0lBBYwFAYIKwYBBQUHAwIGCCsGAQUFBwMB
MA8GA1UdEwEB/wQFMAMBAf8wHAYDVR0OBBUEE2pAdG9yb250b2NyeXB0by5vcmcw
HgYDVR0jBBcwFYATakB0b3JvbnRvY3J5cHRvLm9yZzALBgkqhkiG9w0BAQsDggIB
AJGmZv3TKCwuNafmPUCvvJV6PwdBqYdVX270pLI2IjPa5sE+dDiCrrrH5tVsoUfY
1xAy0eclic3SCu2DdQxicYFIsyN91oyZWljnVuOWDRQoyeGvcwN3FN8WQZ/VnoX/
b4Xtx0D3HsQjLXfzk0AzSXp9TP9/orMR5bkWiqhUhXvlb7XhpZ+p9/8N0D7bjcaJ
74Rn6g3sS+/wKJ0c7h5R3+mRNPW1SecbfQFN/GkgDQxZscvmbRsCG03IRQeYpqt2
M8KA5KXu/H6ZU5XlC6+VI7vf6yWWPf3s8CRBDgfYtI7uRFkfwJLsTBZCOFoyQe+F
CIZZj4lg6f46FHMekbPouw+g2B+2QNdW+fZqdVLAXbuN2xMsVakZn5X9iBfanNmN
t5QH4T81SZb9ZIJSD+L0lKiMw1klbaYYPp2mjwbo42DhsezcJX3TKXhMe3qkYZ3I
E0a9Kq4TmoWAkdycT1oH51wmybwWc3ix7rXbUe8h6KgBEXqJV60ybX7iacrq9WgG
xIr5hnSUEGZtMcdhEA4oD319h+8j/UjXKgWwuuNExpSnARbwQTbPJ/PLD6mQVpHv
jL2S9nbb1r/GmRdzCpHVwLGczUJvwfjAZ8bDCONSGHzuzw8lgpdRpdeWCLfQzXyo
mjh0U8QNpeHEMdQhmnaYa8WJ83DTnO7pwaoYqjeDQ9yM
-----END CERTIFICATE-----
```

**Comment by anonimal** [comment]
Only fixed if assuming #3 is resolved.

**Comment by anonimal** [comment]
This ticket is essentially resolved: bc42f4c mostly resolves my previous comment for satisfying the ticket and 11c0197 specifically resolves this ticket.

OpenSSL is not yet implemented for su3 cert handling *but* with our new crypto_refactor, we may skip out on an openssl su3 re-write all-together and stick with crypto++ because a future user may not want to https reseed - and thus not need an openssl dependency to begin with.

@majestrate, should we close this ticket now? This issue is resolved so, with the knowledge that su3 cert handling will be rewritten *or* handled via crypto_refactor, I vote to close now.

**Comment by anonimal** [comment]
Closing as resolved. Please comment if we should re-open.
